crates/ruff/src/rules/isort/rules/add_required_imports.rs

```diff
@@ -120,7 +120,10 @@ fn add_required_import(
         TextRange::default(),
     );
     if autofix.into() && settings.rules.should_fix(Rule::MissingRequiredImport) {
-        diagnostic.set_fix(Importer::new(python_ast, locator, stylist).add_import(required_import));
+        diagnostic.set_fix(
+            Importer::new(python_ast, locator, stylist)
+                .add_import(required_import, TextSize::default()),
```

Comment:
Shouldn't this use the location of the first non-import statement, or do they need to be inserted even before the other imports?

Reply:
They should be inserted at the top of the file -- since it has to work with `__future__` imports.